### PR TITLE
[CIT-330] Sample collectd configuration file

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

A sample collectd config file that the users can use to enable some basic monitoring collecting the stats of cpu, memory and disk.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
https://cumulocity.atlassian.net/browse/CIT-330

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: contrib/collectd/tedge-collectd.conf [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <Publish "tedge">
+                Host "localhost"
+                Port 1883
+                ClientId "tedge-dm"
+#               QoS 0
+#               Prefix "collectd"
+#               StoreRates true
+#               Retain false
+        </Publish>
+</Plugin>
```
Comment (makr11st) [inline contrib/collectd/tedge-collectd.conf:67]: New line missing.

**Review by makr11st — APPROVED** [review]
Just add new line at the end of the file, but LGTM otherwise.

**Review by didier-wenzek — COMMENTED** [review]
Why a new `contrib` main directory? We already have a `configuration` directory. I would prefer to use the latter.

**Comment by albinsuresh** [comment]
> Why a new `contrib` main directory? We already have a `configuration` directory. I would prefer to use the latter.

I didn't put this in the `configuration` directory as this is more of a sample demo configuration of a third-party tool rather than "our own" configuration. I was hoping that a separate `contrib` directory would make that distinction clear and explicit to the end-user.

**Comment by itsyitsy** [comment]
I also would prefer having no new top-level directory. How about a subdirectory inside configuration?

**Comment by didier-wenzek** [comment]
> I was hoping that a separate `contrib` directory would make that distinction clear and explicit to the end-user.

I don't think that it helps to have so many root directories. On the contrary. Besides, for the end-user, the difference, between a configuration file we support and one we provide, is subtle.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**